### ISSUE TITLE
Consider decoupling dotnet publish-iis from project.json to enable cross publishing

### ISSUE THREAD

**Issue by mikes-gh** [aspnet/IISIntegration#145]
Currently dotnet publish-iis needs dotnet publish to run first. and the default is a post publish entry in project.json.

It would be nice to decouple dotnet publish-iis  from the project.json

**For IIS**
`dotnet publish-iis -o  \\server\approot`
(implicit content entry for web.config)

This would run dotnet publish followed by the current dotnet publish-iis
I realise this can be achieved by running the tools one after the other but I cant see a scenario where I would want to run dotnet publish-iis without dotnet publish first.

**For Kestrel standalone (dev Linux Mac etc)**
`dotnet publish -o c:\localserver\approot`
(never publishes web.config regardless of content entry in project.json)

using the same project.json

UseIISIntegration is already intelligent enough to work out if it is on IIS from the envvariables.

That way I can maintain one project for IIS, OSX, Linux

**Comment by moozzyk** [comment]
> never publishes web.config regardless of content entry in project.json

What's in the `content` section in project.json is being copied blindly. This is done by dotnet cli and I don't think there is a way to prevent this based on the name of the file. 
Postpublish scripts can actually know what platform the project was published for by reading the %publish:Runtime% variable. What you could do is to write a script that checks the target platform and delete the web.config file. I don't think we will add a functionality like this to the publish-iis tool. Having web.config in the publish output is not pretty but it does no cause any harm.
One question I have though is whether it is a real scenario where you create a web app and want to target both Windows and Linux. I thought that when you start working on a web you decide what production environment you want to run your application in and you optimize for it (in this case - you either run publish-iis or not). Because if you want to build for Windows and Linux then you should most likely write web.config for windows and nginx configuration for Linux.

**Comment by mikes-gh** [comment]
@moozzyk 
Publishing to dev and production might be a common scenario. where you might use one project.json.
I think it would be good if dotnet publish-iis was a one step command incorporating dotnet publish.  It just seems more natural to me.  Otherwise you  have to explain that dotnet publish-iis doesn't publish the project but only the web.config.  It seems a misnomer.

Or it may even be better to just forget about the publish-iisand just have a web.config template in vstudio/vscode that creates the webconfig and just use normal publish.

**Comment by GuardRex** [comment]
> What you could do is to write a script that checks the target platform and delete the web.config file.

... or do what I did: Clone the tool and hack away. It's easy to mod to one's liking. I hacked it to set the default for log output to `true` and name the log file prefix to the application name in 15 minutes! :smile:

> It seems a misnomer.

:+1: It isn't named according to what the tool actually does, which would be something like "configure-web.config," "configure-for-iis," or something along those lines.
